v2/pkg/operators/common/dsl/dsl.go

```diff
@@ -50,6 +50,13 @@ var invalidDslFunctionMessageTemplate = "%w. correct method signature %q"
 
 var dslFunctions map[string]dslFunction
 
+var (
+	// FunctionNames is a list of function names for expression evaluation usages
+	FunctionNames []string
```

Comment:
Is extracting the keys of the `HelperFunctions` map really necessary? We are basically duplicating the values...

Couldn't we simply rather use https://pkg.go.dev/golang.org/x/exp/maps#Keys or something?

Reply:
this is what we were previously doing and it was allocating around 6gb of memory during a large scan. it is very good for performance and allocation if we pre-allocate and extract these values before using them